geniza/corpus/management/commands/export_metadata.py

```diff
@@ -1,9 +1,106 @@
+import os
 import time
 
+from django.conf import settings
 from django.core.management.base import BaseCommand
 from django.utils import timezone
+from git import Repo
 
-from geniza.corpus.metadata_export import DocumentExporter
+from geniza.corpus.metadata_export import PublicDocumentExporter
+
+
+class MetadataExportRepo:
+    local_path_settings_key = "METADATA_REPO_LOCAL_PATH"
+    remote_url_settings_key = "METADATA_REPO_REMOTE_URL"
+    default_remote_url_org = "https://github.com/Princeton-CDH"
+
+    repo_name = "test-geniza-metadata"
+    repo_dir_data = "data"
+    ext_csv = ".csv"
+
+    def __init__(self, local_path=None, remote_url=None):
+        self._local_path = local_path
+        self._remote_url = remote_url
+        self._repo = None
+
+    @property
+    def local_path(self):
+        if not self._local_path:
+            if hasattr(settings, self.local_path_settings_key):
+                self._local_path = getattr(settings, self.local_path_settings_key)
+            else:
+                self._local_path = os.path.abspath(
+                    os.path.join(settings.BASE_DIR, "..", self.repo_name)
+                )
+        return self._local_path
+
+    @property
+    def remote_url(self):
+        if not self._remote_url:
+            if hasattr(settings, self.remote_url_settings_key):
+                self._remote_url = getattr(settings, self.remote_url_settings_key)
+            else:
+                self._remote_url = os.path.join(
+                    self.default_remote_url_org, self.repo_name
+                )
+        return self._remote_url
+
+    @property
```

Comment:
could use `cached_property` from functools instead of the `self._repo` logic here

Reply:
🤯

Reply:
Good example of a Python innovation I just haven't picked up yet

Reply:
it's so hard to keep up with all the new things! for a while I was using a third-party cached property package, so I was delighted when I figured out there was a built in version added at some point (I forget what version of python)